cardano-db-sync/src/Cardano/DbSync/LedgerState.hs

```diff
@@ -325,16 +367,35 @@ applyBlock env blk = do
               { Generic.neEpoch = ledgerEpochNo env newState
               , Generic.neIsEBB = isJust $ blockIsEBB blk
               , Generic.neAdaPots = maybeToStrict $ getAdaPots newState
-              , Generic.neEpochUpdate = Generic.epochUpdate (clsState newState)
+              , Generic.neEpochUpdate = Generic.epochUpdate newState
               }
 
+    applyToEpochHeight :: Bool -> Bool -> EpochHeight -> EpochHeight
+    applyToEpochHeight True _ _ = EBBHeight
+    applyToEpochHeight _ True _ = Height 0
+    applyToEpochHeight _ _ (Height n) = Height (n + 1)
+    applyToEpochHeight _ _ GenesisHeight = Height 0
+    applyToEpochHeight _ _ EBBHeight = Height 0
+
+    stakeSliceSize :: Word64
+    stakeSliceSize = 2000
```

Comment:
Sorry for dropping in, but is there potential for this value to be configurable by users - via config or startup arg?

Would be good to be able to tune stake chunks from consumer pov to be larger/smaller depending on instance performance and actual amount of stakes

Reply:
Not sure making this configurable is such a good idea. Its too easy for a user the configure it to a value that causes problems, but the problems will not be easily related to this value. There is also no advantage for to either making it slower or faster.

Reply:
Values less than 100 on mainnet could cause issues, since with ~21000 blocks per epoch, this gives us max 2100000 delegations. Maybe it's even worth having a sanity check that this value is bigger than `expected_blocks / size_of_delegations` or it could be adjusted per epoch to a big enough value if that's not the case.

Reply:
Right, my primary concern is that other than mainnet and testnet, there are also small networks run by community (guild network - 1 hour, and many run short plutus networks) that have very low amount of blocks per epoch, but can have:
- very low amount of stake initially , as network isn't used by many
- very high amount of stakes as part of burst tests
If we think value of 2000 is safe enough even for small networks, we might be OK.
I understand that it could be misused, but if above becomes a concern - maybe it only be exposed as an advanced option , that not every dbsync user should touch?

Reply:
It might make sense to parameterize this value on expected blocks per epoch, but it does not make sense to expose this to users.

@kderme It would be good to think about the issue @rdlrt raises and also think about some of the testnets we run which have a very low quality factor (ie large numbers of missing blocks).

Reply:
Ok changed this. The slice size is now `max 2000 defaultEpochSliceSize`. `defaultEpochSliceSize` is a value big enough to cover everything even if only 20% of the expected blocks are produced in the epoch.

So we could even make 2000 configurable, since if the user provides a very small value it will be orewriten. But I'd leave this for another pr.